internal/integrationtest/compile/compile_part_3_test.go

```diff
@@ -0,0 +1,353 @@
+// This file is part of arduino-cli.
+//
+// Copyright 2022 ARDUINO SA (http://www.arduino.cc/)
+//
+// This software is released under the GNU General Public License version 3,
+// which covers the main part of arduino-cli.
+// The terms of this license can be found at:
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+
+package compile_part_1_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/arduino/arduino-cli/internal/integrationtest"
+	"github.com/arduino/go-paths-helper"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+/*func TestCompileWithFullyPrecompiledLibrary(t *testing.T) {
```

Comment:
maybe it would make sense to find a lib to be replaced in this PR since we are refactoring the tests.. This one should do: https://github.com/arduino-libraries/Arduino_Braccio_plusplus

Reply:
I don't know, it was a very weird particular case, if we can find the old artifacts maybe we can re-upload the library just for testing.

Reply:
let's try to manually install the library using `--zip path` flag

Reply:
Updated the test using an archived version of the library. I could modify it later if we find better solutions